packages/fxa-settings/src/pages/Pair/Index/index.tsx

```diff
@@ -0,0 +1,178 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useEffect } from 'react';
+import { Link, useNavigate } from '@reach/router';
+import { RouteComponentProps } from '@reach/router';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { usePageViewEvent } from '../../../lib/metrics';
+import { useFtlMsgResolver } from '../../../models';
+import Banner, { BannerType } from '../../../components/Banner';
+import CardHeader from '../../../components/CardHeader';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { ENTRYPOINTS, REACT_ENTRYPOINT } from '../../../../src/constants';
+import { HeartsVerifiedImage } from '../../../components/images';
+
+type PairProps = {
+  error?: string;
+  entryPoint?: ENTRYPOINTS;
+  onSubmit: Function; // navigates to `about:preferences` whatever the given broker does that.
+};
+export const viewName = 'pair';
+
+const Pair = ({
+  error,
+  entryPoint,
+  onSubmit,
+}: PairProps & RouteComponentProps) => {
+  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedQRCodeLabel = ftlMsgResolver.getMsg(
+    'pair-qr-code-aria-label',
+    'QR code'
+  );
+  const navigate = useNavigate();
+  // TODO: Recreate the QR code logic which previously existed in the content-server.
+  // Probably after that we can remove the fallback styles for the QR code div.
+
+  // TODO: check if we are either using Firefox Desktop ,or if the UA has the `supportPairing` capability
+  const isSupported = () => true;
+
+  // TODO: get the account and check if it is the default account.
+  const isDefaultAccount = () => false;
+
+  // TODO: check if the account is either not verified, or has a falsy value for session token.
+  const accountIsNotVerifiedOrHasNoSessionToken = () => false;
+
+  useEffect(() => {
+    // This will just run at page load.
+    if (!isSupported()) {
+      navigate('/pair/unsupported');
+    }
+    if (isDefaultAccount()) {
+      // we have historically needed the "forceView" option to prevent a loop of redirects.
+      navigate('/connect_another_device', { state: { forceView: true } });
+    }
+    if (accountIsNotVerifiedOrHasNoSessionToken()) {
+      navigate('/signin');
+    }
+  }, [navigate]);
+
+  const showQRCode =
+    entryPoint === ENTRYPOINTS.FIREFOX_MENU_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_PREFERENCES_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_SYNCED_TABS_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_TABS_SIDEBAR_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_FX_VIEW_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_TOOLBAR_ENTRYPOINT;
```

Comment:
I think the only one that this isn't checking is `FIREFOX_IOS_OAUTH_ENTRYPOINT = 'ios_settings_manage',` enum value.
Not sure if it'd be easier (but possibly riskier) to check if `entryPoint !== ENTRYPOINTS.FIREFOX_IOS_OAUTH_ENTRYPOINT` or you could possibly do `Array#includes()`, although not sure that is any better:

```js
const showQRCode = [
  ENTRYPOINTS.FIREFOX_MENU_ENTRYPOINT,
  ENTRYPOINTS.FIREFOX_PREFERENCES_ENTRYPOINT,
  ENTRYPOINTS.FIREFOX_SYNCED_TABS_ENTRYPOINT,
  ENTRYPOINTS.FIREFOX_TABS_SIDEBAR_ENTRYPOINT,
  ENTRYPOINTS.FIREFOX_FX_VIEW_ENTRYPOINT,
  ENTRYPOINTS.FIREFOX_TOOLBAR_ENTRYPOINT,
].includes(entryPoint);
```